### PR TITLE
[ROCKETMQ-359] Replace slf4j api used in RocketMQ with InternalLogger

### PR DESCRIPTION
**Pull request by lindzh**
## What is the purpose of the change

As logging isolation has been fix with logging component, it is necessary to replace slf4j api used in RocketMQ with InternalLogger now!

## Brief changelog

1. Replace slf4j api used in RocketMQ.
2. Fix ClientLogger with InnerLoggerFactory.
3. Add NettyLogger adapter using InternalLogger for remoting.

## Verifying this change

Review the source code and run the unit/it tests

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/ROCKETMQ/issues/) filed for the change (usually before you start working on it). Trivial changes like typos do not require a JIRA issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ROCKETMQ-XXX] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [x] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by vongosling** [comment]
Could we split the pr into the two parts, one is for log component polish, another is for substitute globally.

**Comment by lindzh** [comment]
@vongosling , Thx, in the next commit,it will be split into two parts.

**Comment by lindzh** [comment]
Logging component has been spilted into This PR https://github.com/apache/rocketmq/pull/224.

**Comment by lindzh** [comment]
@vongosling @justinmclean Logging has been split to logging component and this PR, logging component has been merged to develop , now this PR has merged develop and changed title for next replacement purpose, please help a review.

File: remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyLogger.java [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+        public NettyBridgeLogger(String name) {
+            logger = InternalLoggerFactory.getLogger(name);
+        }
+
+        @Override
+        public String name() {
+            return logger.getName();
+        }
+
+        // FIXME: 2018/3/2
```
Comment (zhouxinyu) [inline remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyLogger.java]: How to fix this issue?

Comment (lindzh) [reply remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyLogger.java]: Netty Logger has been hard coded to use ERROR log level.  I think this is not a good idea.  I hope to fix this in next PR.

Comment (lindzh) [reply remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyLogger.java]: Fixme has been deleted.

File: remoting/src/main/java/org/apache/rocketmq/remoting/common/ServiceThread.java [hunk 2]
```diff
@@ -16,14 +16,15 @@
  */
 package org.apache.rocketmq.remoting.common;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+
+import org.apache.rocketmq.logging.InternalLogger;
```
Comment (vongosling) [inline remoting/src/main/java/org/apache/rocketmq/remoting/common/ServiceThread.java]: Manual replace?

Comment (lindzh) [reply remoting/src/main/java/org/apache/rocketmq/remoting/common/ServiceThread.java]: Yes

**Comment by zhouxinyu** [comment]
Merged thanks.

**Comment by ShiXulu** [comment]
but why? why rocketmq didn't use slf4j?

**Pull request merged**